File: easy/real_fake.hs
import System.Environment (getArgs)
import Data.Char (isDigit)

toDigits  :: Integer -> [Integer]
toDigits x | x < 0     = [-1]
           | x < 10    = [x]
           | otherwise = toDigits (div x 10) ++ [mod x 10]

toDigitsRev  :: Integer -> [Integer]
toDigitsRev x | x < 0  = [-1]
              | otherwise = reverse (toDigits x)

dobSec          :: [Integer] -> [Integer] -> [Integer]
dobSec xs []     = xs
dobSec xs [x]    = xs ++ [x]
dobSec xs (y:ys) = dobSec (xs ++ [y, 2 * head ys]) (tail ys)

isValid  :: Integer -> String
isValid x | mod (sum (dobSec [] (toDigitsRev x))) 10 == 0 = "Real"
          | otherwise                                     = "Fake"

main :: IO ()
main = do
    [inpFile] <- getArgs
    input <- readFile inpFile
    putStr . unlines . map (isValid . read . filter isDigit) $ lines input

File: easy/real_fake.lua
for line in io.lines(arg[1]) do
  local a, su = {}, 0
  for i in line:gmatch("%d") do
    a[#a+1] = tonumber(i)
  end
  for i = #a-1, 1, -2 do
    a[i] = 2*a[i]
  end
  for i = 1, #a do
    su = su + a[i]
  end
  print((su%10 == 0) and "Real" or "Fake")
end
